### PR TITLE
doc: zbus: release notes and migration guide Zephyr 3.6

### PR DESCRIPTION
**Pull request by rodrigopex**
Modify the release notes and migration guide for zbus.

### CODE REVIEW

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -1229,10 +1229,26 @@ Libraries / Subsystems
 
 * ZBus
 
+  * Replace mutexes with semaphores to lock channels and implement the Highest Locker Protocol (HLP)
+    priority boost for the zbus operations. This feature avoids priority inversions and preemptions,
+    making the VDED delivery process faster and more consistent. More details at
+    https://github.com/zephyrproject-rtos/zephyr/pull/63183.
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.6.rst:1235]: This needs to be a proper reference, see examples in the migration guide.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.6.rst (hunk 1 above)
Comment (rodrigopex) [reply doc/releases/release-notes-3.6.rst]: Done. I have added the reference to the other items as well. Thanks @henrikbrixandersen.

**Review by rodrigopex — COMMENTED** [review]
Add the proper reference to all the items from release notes and migration guides.

**Review by henrikbrixandersen — DISMISSED** [review]
Thanks!

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.6.rst [hunk 2]
```diff
@@ -1229,10 +1229,25 @@ Libraries / Subsystems
...
+  * Replace mutexes with semaphores to lock channels and implement the Highest Locker Protocol (HLP)
+    priority boost for the zbus operations. This feature avoids priority inversions and preemptions,
+    making the VDED delivery process faster and more consistent. (:github:`63183`)
+
+  * Fixed documentation for :c:func:`zbus_chan_add` and :c:func:`zbus_chan_rm` adding the timeout
+    argument. (:github:`65544`)
+
+  * Fixed warning when mixing C and C++ files using zbus. (:github:`65222`)
+
+  * :kconfig:option:`ZBUS_CHANNEL_DEFINE` macro is now compatible with C++. (:github:`65196`)
```
[the remark this answers is not in the document]
Comment (rodrigopex) [reply doc/releases/release-notes-3.6.rst]: Sorry, this is not a kconfig. It is a macro. Fixed.

File: doc/releases/migration-guide-3.6.rst [hunk 3]
```diff
@@ -587,10 +587,13 @@ Other Subsystems
   respective ``reset-gpios``. This has been fixed so those signals now have to
   be flagged as :c:macro:`GPIO_ACTIVE_LOW` in the devicetree. (:github:`64800`)
 
-* The :kconfig:option:`ZBUS_MSG_SUBSCRIBER_NET_BUF_DYNAMIC`
-  and :kconfig:option:`ZBUS_MSG_SUBSCRIBER_NET_BUF_STATIC`
-  zbus options are renamed. Instead, the new :kconfig:option:`ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_DYNAMIC`
-  and :kconfig:option:`ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_STATIC` options should be used.
+* The ``ZBUS_MSG_SUBSCRIBER_NET_BUF_DYNAMIC`` and ``ZBUS_MSG_SUBSCRIBER_NET_BUF_STATIC`` zbus
+  options are renamed. Instead, the new :kconfig:option:`ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_DYNAMIC` and
+  :kconfig:option:`ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_STATIC` options should be used. (:github:`65632`)
```
[the remark this answers is not in the document]
Comment (rodrigopex) [reply doc/releases/migration-guide-3.6.rst]: done

**Review by rodrigopex — COMMENTED** [review]
Solve the `CONFIG_` prefix and `:c:macro` missing.

**Review by MaureenHelm — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**